test/unit/models/tasks.test.js

```diff
@@ -70,6 +71,63 @@ describe("tasks", function () {
       }
     });
   });
+
+  describe("fetchTasks", function () {
+    beforeEach(async function () {
+      const tasksPromise = tasksData.map(async (task) => {
+        await tasks.updateTask(task);
+      });
+      await Promise.all(tasksPromise);
+    });
+
+    it("should fetch all tasks when dev is false and no status is passed to the fetchTasks function", async function () {
+      const result = await tasks.fetchTasks();
+
+      expect(result).to.have.length(tasksData.length);
+      result.forEach((task) => {
+        const sameTask = tasksData.find((t) => t.title === task.title);
+        expect(task).to.contain.all.keys(sameTask);
+      });
+    });
+
+    it("should fetch all tasks when dev is false but status is passed to the fetchTasks function", async function () {
+      const status = TASK_STATUS.ASSIGNED;
+      const result = await tasks.fetchTasks(false, status);
+
```

Comment:
why we not filtering when dev=false??

Reply:
That's the requirement. To flag this feature so that the filtering only happens in dev mode.